google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/JsonToProtoMessage.java

```diff
@@ -0,0 +1,406 @@
+              "JSONObject does not have the required field " + currentScope + ".");
+        } else {
+          continue;
+        }
+      }
+      matchedFields++;
+      if (!field.isRepeated()) {
+        fillField(
+            protoMsg,
+            field,
+            json,
+            jsonLowercaseNameToActualName.get(lowercaseFieldName),
+            currentScope,
+            allowUnknownFields);
+      } else {
+        fillRepeatedField(
+            protoMsg,
+            field,
+            json,
+            jsonLowercaseNameToActualName.get(lowercaseFieldName),
+            currentScope,
+            allowUnknownFields);
+      }
+    }
+    if (matchedFields == 0 && topLevel) {
+      throw new IllegalArgumentException(
+          "There are no matching fields found for the JSONObject and the protocol buffer descriptor.");
+    }
+    return protoMsg.build();
+  }
+
+  /**
+   * Fills a non-repetaed protoField with the json data.
+   *
+   * @param protoMsg The protocol buffer message being constructed
+   * @param fieldDescriptor
+   * @param json
+   * @param actualJsonKeyName Actual key name in JSONObject instead of lowercased version
+   * @param currentScope Debugging purposes
+   * @param allowUnknownFields Ignores unknown JSON fields.
+   * @throws IllegalArgumentException when JSON data is not compatible with proto descriptor.
+   */
+  private static void fillField(
+      DynamicMessage.Builder protoMsg,
+      FieldDescriptor fieldDescriptor,
+      JSONObject json,
+      String actualJsonKeyName,
+      String currentScope,
+      boolean allowUnknownFields)
+      throws IllegalArgumentException {
+
+    switch (fieldDescriptor.getType()) {
+      case BOOL:
+        try {
+          protoMsg.setField(fieldDescriptor, new Boolean(json.getBoolean(actualJsonKeyName)));
+        } catch (JSONException e) {
+          throw new IllegalArgumentException(
+              "JSONObject does not have a boolean field at " + currentScope + ".");
+        }
+        break;
+      case BYTES:
+        try {
+          protoMsg.setField(fieldDescriptor, json.getString(actualJsonKeyName).getBytes());
+        } catch (JSONException e) {
+          throw new IllegalArgumentException(
+              "JSONObject does not have a string field at " + currentScope + ".");
+        }
+        break;
+      case INT64:
+        try {
+          java.lang.Object val = json.get(actualJsonKeyName);
+          if (val instanceof Byte) {
+            protoMsg.setField(fieldDescriptor, new Long((Byte) val));
+          } else if (val instanceof Short) {
+            protoMsg.setField(fieldDescriptor, new Long((Short) val));
+          } else if (val instanceof Integer) {
+            protoMsg.setField(fieldDescriptor, new Long((Integer) val));
+          } else if (val instanceof Long) {
+            protoMsg.setField(fieldDescriptor, new Long((Long) val));
+          } else {
+            throw new IllegalArgumentException(
```

Comment:
Does this block equals to the following code?:

try {
        return Long.parseLong(json.getAsString());
} catch (Exception e) {
       throw new IllegalArgumentException();
}

Reply:
I'm using org.json.JSONObject which I don't think has that method. Should I be using something else?